src/Security/LogInAuthenticationHandler.php

```diff
@@ -63,6 +64,11 @@ public function authenticateRequest(HTTPRequest $request)
 
     public function logIn(Member $member, $persistent = false, HTTPRequest $request = null)
     {
+        if ($request == null && !Controller::has_curr()) {
+            throw new InvalidArgumentException("No HTTPRequest is provided and there is no controller either.");
+        }
+        $request = Controller::curr()->request;
```

Comment:
Couple more suggestions on this logic:

- We should only try to re-fetch the $request object if it isn't provided directly to the method.
- The error message should try to convey what the consequence of the missing request object is
- Prefer using getter methods over directly accessing object properties (the `request` property is `protected`, so this seems only to be working currently because of Silverstripe ✨magic✨)

```suggestion
        // Fall back to retrieving request from current Controller if available
        if ($request === null) {
            if (!Controller::has_curr()) {
                throw new InvalidArgumentException("Authentication with SessionManager enabled requires an active HTTPRequest.");
            }
            
            $request = Controller::curr()->getRequest();
        }
```